server/src/main/java/io/deephaven/server/console/ScopeTicketResolver.java

```diff
@@ -58,29 +51,29 @@ public SessionState.ExportObject<Flight.FlightInfo> flightInfoFor(
         // there is no mechanism to wait for a scope variable to resolve; require that the scope variable exists now
         final String scopeName = nameForDescriptor(descriptor, logId);
 
-        final ScriptSession gss = scriptSessionProvider.get();
-        final Flight.FlightInfo flightInfo =
-                gss.getExecutionContext().getUpdateGraph().sharedLock().computeLocked(() -> {
-                    Object scopeVar = gss.getVariable(scopeName, null);
-                    if (scopeVar == null) {
-                        throw Exceptions.statusRuntimeException(Code.NOT_FOUND,
-                                "Could not resolve '" + logId + ": no variable exists with name '" + scopeName + "'");
-                    }
-                    if (scopeVar instanceof Table) {
-                        scopeVar = authorization.transform(scopeVar);
-                        return TicketRouter.getFlightInfo((Table) scopeVar, descriptor, flightTicketForName(scopeName));
-                    }
+        QueryScope queryScope = ExecutionContext.getContext().getQueryScope();
+        Object scopeVar = queryScope.readParamValue(scopeName, null);
+        if (scopeVar == null) {
+            throw Exceptions.statusRuntimeException(Code.NOT_FOUND,
+                    "Could not resolve '" + logId + ": no variable exists with name '" + scopeName + "'");
+        }
+        if (!(scopeVar instanceof Table)) {
+            throw Exceptions.statusRuntimeException(Code.NOT_FOUND,
+                    "Could not resolve '" + logId + "': no variable exists with name '" + scopeName + "'");
```

Comment:
This is a lie.

Reply:
What do you mean by "doesn't seem constructive"? It seems like pretty valuable debugging information to me.